Example/Tests/GINICameraViewControllerTests.swift

```diff
@@ -69,5 +76,81 @@ final class CameraViewControllerTests: XCTestCase {
                       "multipageReviewBackgroundView should not be hidden after capture the second picture")
         
     }
+    
+    func testPickerCompletionBlockWhenNoErrors() {
+        let documents = [GiniImageDocument(data: imageData, imageSource: .external)]
+        let expect = expectation(description: "Document validation finishes")
+
+        vc.documentPicker(DocumentPickerCoordinator(), didPick: documents, from: .gallery) { error, _ in
+            expect.fulfill()
+            XCTAssertNil(error, "Completion block should not return an error when pciking one image")
+        }
+        
+        waitForExpectations(timeout: 2, handler: nil)
+    }
+    
+    func testPickerCompletionBlockWhenNoErrorsWithDeprecatedInit() {
+        let documents = [loadImageDocument(withName: "invoice")]
+        let expect = expectation(description: "Document validation finishes")
+        vc = CameraViewController(success: {_ in}, failure: {_ in})
+        vc.documentPicker(DocumentPickerCoordinator(), didPick: documents, from: .gallery) { error, _ in
+            expect.fulfill()
+            XCTAssertNil(error, "Completion block should not return an error when using the deprecated initializer")
+        }
+        
+        waitForExpectations(timeout: 2, handler: nil)
+    }
+    
+    func testPickerCompletionBlockWhenFailedPDF() {
+        let failedPDF = loadPDFDocument(withName: "testPDF")
+        failedPDF.error = DocumentValidationError.pdfPageLengthExceeded
+        let documents = [failedPDF]
+        let expect = expectation(description: "Document validation finishes")
+        
+        vc.documentPicker(DocumentPickerCoordinator(), didPick: documents, from: .gallery) { error, _ in
+            expect.fulfill()
+            XCTAssertNil(error, "Completion block should not return an error form outside when it is a PDF")
```

Comment:
_from*_

Also, I don't understand - why not return an error?

Reply:
Because when there is a PDF, there are no validation from outside (since it can only be one document not several documents.)